ui/raidboss/data/07-dt/raid/r2s.ts

```diff
@@ -96,7 +99,25 @@ const triggerSet: TriggerSet<Data> = {
       id: 'R2S Headmarker Cone Tankbuster',
       type: 'HeadMarker',
       netRegex: { id: headMarkerData.tankLaser, capture: true },
-      response: Responses.tankCleave(),
+      delaySeconds: (data, matches) => {
+        data.tankLaserCollect.push(matches);
+        return data.tankLaserCollect.length === 2 ? 0 : 0.1;
+      },
+      alertText: (data, _matches, output) => {
+        if (data.tankLaserCollect.length === 0)
+          return;
```

Comment:
I don't think this early exit will ever be triggered -- the first time the trigger is fired, the `delaySeconds` func will push the first match to `data.tankLaserCollect`, so `data.tankLaserCollect.length` will never be 0 by this point?  Maybe instead:

```suggestion
        if (data.tankLaserCollect.length < 2)
          return;
```

Reply:
Let me check, I think it can because the tankLaserCollect is cleared in the run() function. I modeled this after the way that other single-trigger collect things work: the collect is done in delay seconds, then the alert function checks the length, while the run function clears the array. Is that not correct?

Reply:
> Let me check, I think it can because the tankLaserCollect is cleared in the run() function. I modeled this after the way that other single-trigger collect things work: the collect is done in delay seconds, then the alert function checks the length, while the run function clears the array. Is that not correct?

Key things to note here are that (1) this trigger will fire twice; (2) the functions will be executed sequentially, e.g., `delaySeconds` -> `alertText` -> `run` each time the trigger fires, unless there is a `condition` that prevents those functions from being executed.  In other words, the `run` function will be executed after `alertText` both times the trigger is run. 

As this is currently written, `delaySeconds` will push the headmarker target into `data.tankLaserCollect`, whose `.length === 1` when `alertText` runs (so the early exit is never triggered), and `run` then clears the array.  When the second headmarker line is received, the exact same process will repeat, with a 0.1 second delay.

(NB: We often have separate collector vs. output triggers.  It's entirely possible to combine both in a single trigger like you've done, but it just need to take into account the above rules.)

Reply:
(Opinion only, non-binding)

As I noted on the Beat 2 spread changes, unless there's a specific reason to do single-trigger collect-call-clear constructions, I do think we're better off collecting in a separate trigger, and then changing the call trigger to include a 0.1s delay, and an early return within the `alertText` if `data.tankLaserCollect.length` is not 2. Obviously this does have the downside of permanently including a 0.1s delay for the call trigger, but that's still going to give users More Than Five seconds to react, and that's above our standard four for AoEs.

Reply:
Sure I can look into that. I personally found same trigger easy to read but it does seem less common.

Reply:
> Sure I can look into that. I personally found same trigger easy to read but it does seem less common.

fwiw, I agree with @JLGarber -- I think it would be cleaner if collect vs. call are handled by separate triggers (with cleanup handled in the `run` block of the call trigger). 

(And, as an aside/for future reference, having separate triggers is almost always easier to debug in raidemulator if there's a problem somewhere.  Based on personal experience. 🤣 ).

Reply:
Ah, that is a good point. While we don't precisely want to write for the raid emulator, I do think it's worthwhile to deliberately make it easier where it's basically free. (As another example where I do this almost for free, see <https://github.com/OverlayPlugin/cactbot/blob/d375dce10384a2b0a568b41dda115672da577128/ui/raidboss/data/03-hw/trial/sephirot-ex.ts#L111>. This construction avoids having this trigger activate every three seconds during the encounter, which is otherwise incredibly noisy in the emulator.)